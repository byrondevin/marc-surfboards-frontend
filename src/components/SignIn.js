
import axios from "axios";


export default function SignIn() {




    //signIn function. Triggered when signup form is subitted
    async function login (e) {

        //prevent default form redirection
        e.preventDefault();

        //try axios.post to sign in
        try{
            //login with post. recieve JWT token in return at user.data.token
            const { user } = await axios.post(`api/sign-in`, {email: e.target[0].value, password: e.target[1].value});
            console.log("USER");
            console.log(user);

            //set JWT token in session storage
            sessionStorage.setItem('JWT', user.data.token);
            let JWT = sessionStorage.getItem('JWT');

            console.log("JWT");
            console.log(JWT);

            //Inform user of successfull login
            alert("Login successful");

            // //redirect back to homepage
            window.location.replace('/');


        }
        //catches axios post errors and prints them out
        catch(e){

            console.log("ERROR LOGGING IN")
            console.log(e)
        
        }

        


    }



  return (
    <>

    {/* SIGNIN CONTENT CONTAINER */}
      <div className="SignIn container-fluid">

            {/* Heading 1 row an col */}
            <div className="row text-center">
                <div className="col">
                    <h1>Log in</h1>
                </div>
            </div>

            {/* Form row */}
            <div className="row justify-content-center text-center">
                
                {/* Form col. Responsive */}
                <div className="col-sm-6 m-5">

                    {/* Sign up form. Submit triggers signin(). Email & Password fields.  */}
                    <form onSubmit={login}>

                        {/* Email field */}
                        <div className="form-group">
                            <label for="exampleInputEmail1">Email address</label>
                            <input type="email" className="form-control" id="exampleInputEmail1" name="exampleInputEmail1" aria-describedby="emailHelp" placeholder="Enter email"/>
                            <small id="emailHelp" className="form-text text-muted">We'll never share your email with anyone else.</small>
                        </div>

                        {/* Password field */}
                        <div className="form-group">
                            <label for="exampleInputPassword1">Password</label>
                            <input type="password" className="form-control" id="exampleInputPassword1" name="exampleInputPassword1" placeholder="Password"/>
                        </div>

                        {/* Submit button to log in */}
                        <button type="submit" className="btn btn-primary mt-4">Sign In</button>
                
                    </form>
                </div>
            </div>
        </div>
    </>
  )
}
