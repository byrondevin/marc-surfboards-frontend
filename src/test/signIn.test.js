import SignIn from '../components/SignIn';
import React from 'react';
import renderer from 'react-test-renderer'; 
import "@testing-library/jest-dom";
import axios from "axios";



//fetch users from db test
test('sign in', async () => {
  
  let user = await axios.post(`http://localhost:5000/sign-in`, {email: "e@e", password: "e"});
  expect(user.data.token).to.be.defined();
});




//sign-in page snapshot test

test('signIn matches snapshot', () => {
    const tree = renderer
  
      .create(<SignIn />)
      .toJSON()
      expect(tree).toMatchSnapshot();
    
  });

