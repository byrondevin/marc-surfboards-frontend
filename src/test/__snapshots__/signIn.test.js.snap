// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signIn matches snapshot 1`] = `
<div
  className="SignIn container-fluid"
>
  <div
    className="row text-center"
  >
    <div
      className="col"
    >
      <h1>
        Log in
      </h1>
    </div>
  </div>
  <div
    className="row justify-content-center text-center"
  >
    <div
      className="col-sm-6 m-5"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            for="exampleInputEmail1"
          >
            Email address
          </label>
          <input
            aria-describedby="emailHelp"
            className="form-control"
            id="exampleInputEmail1"
            name="exampleInputEmail1"
            placeholder="Enter email"
            type="email"
          />
          <small
            className="form-text text-muted"
            id="emailHelp"
          >
            We'll never share your email with anyone else.
          </small>
        </div>
        <div
          className="form-group"
        >
          <label
            for="exampleInputPassword1"
          >
            Password
          </label>
          <input
            className="form-control"
            id="exampleInputPassword1"
            name="exampleInputPassword1"
            placeholder="Password"
            type="password"
          />
        </div>
        <button
          className="btn btn-primary mt-4"
          type="submit"
        >
          Sign In
        </button>
      </form>
    </div>
  </div>
</div>
`;
